src/utils/cli/interactive_cli.py

```
@@ -261,32 +278,35 @@ def _cmd_vim(self):
+        # Parse the editor command with shlex to support flags
+        editor_cmd = shlex.split(editor, posix=(os.name != 'nt'))
+        # Normalize the executable token on Windows to remove surrounding quotes
+        if os.name == 'nt' and editor_cmd:
+            editor_cmd[0] = editor_cmd[0].strip('"').strip("'")
+        # End-if
 
-            # Read back user input
-            with open(tmp_path, 'r', encoding='utf-8') as f:
-                new_content = f.read().strip()
+        editor_cmd.append(staged_file)
+        subprocess.call(editor_cmd)
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)